invenio_app_ils/records/api.py

```diff
@@ -53,7 +74,112 @@ def create(cls, data, id_=None, **kwargs):
+    def language_relation():
+        """Get language relation type."""
+        return current_app.config["LANGUAGE_RELATION"]
+
+    @property
+    def editions(self):
+        """Get related editions."""
+        return self._get_related(self.edition_relation())
+
+    @property
+    def languages(self):
+        """Get related languages."""
+        return self._get_related(self.language_relation())
+
+    def node(self, relation_type):
+        """Get PIDNode for this record."""
+        return PIDNode(pid=self.pid, relation_type=relation_type)
+
+    def add_related_edition(self, record):
+        """Add record as a related edition record."""
+        relation = self.edition_relation()
+        self.add_related(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def remove_related_edition(self, record):
+        """Remove related edition record."""
+        relation = self.edition_relation()
+        self.remove_related(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def add_related_language(self, record):
+        """Add record as a related language record."""
+        relation = self.language_relation()
+        self.add_related(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def remove_related_language(self, record):
+        """Remove related language record."""
+        relation = self.language_relation()
+        self.remove_related(
+            record,
+            relation,
+            parent_node=self._parent_node(relation)
+        )
+
+    def add_related(self, record, relation_type, parent_node=None):
+        """Add related record with a relation."""
+        record_node = record.node(relation_type)
+        if record_node.is_parent or record_node.is_child:
+            raise RelatedRecordError(
+                "failed to add related record because it already has relations"
+            )
+        if parent_node is None:
+            parent_node = self.node(relation_type)
+        parent_node.insert_child(record.pid)
+
+    def remove_related(self, record, relation_type, parent_node=None):
+        """Remove related record with a relation."""
+        record_node = record.node(relation_type)
+        if record_node.is_parent or record_node.is_child:
+            raise RelatedRecordError(
+                "failed to remove related record - it already has relations"
+            )
+        if parent_node is None:
+            parent_node = self.node(relation_type)
+        parent_node.remove_child(record.pid)
+
+    def _parent_node(self, relation_type):
+        """Get the parent node with the given relation_type."""
+        node = self.node(relation_type)
+        parents = list(node.parents)
+        is_child = len(parents) > 0
+        if is_child:
+            return PIDNode(pid=parents[0], relation_type=relation_type)
```

Comment:
Is the first element in the parents array the nearest parent? Also I see that there is also the option to have ordered pid relations, would this be helpful for multipart monographs?

Reply:
There should never be a case where there are more than 1 parent so it should probably raise an exception if that happens.

Good idea! Yes, the ordering could indeed be helpful for series.